src/main/java/com/aws/greengrass/lifecyclemanager/GenericExternalService.java

```diff
@@ -366,6 +386,10 @@ private synchronized void handleRunScript() throws InterruptedException {
             }
         }, lifecycleProcesses);
 
+        if (result.getRight() != null) {
+            systemResourceController.addComponentProcess(this, result.getRight().getProcess());
+        }
+
```

Comment:
Same here, would be better done on line 400, i.e. `else` -> `else if (result.getRight() != null)`